Translate `library/collections.po`

mattwang44 opened the pull request:
fix #111

mattwang44 committed:
feat: translate `library/collections.po`

toto6038 commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:25
 msgid "factory function for creating tuple subclasses with named fields"
-msgstr "用來建立一個欄位擁有名字的 tuple 子類別的函數"
+msgstr "用來建立其欄位擁有名字的 tuple 子類別的產生函數"
```
用來建立具名欄位的 tuple 子類別的產生函數
參見第1043行

toto6038 commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:30
 msgid "dict subclass that calls a factory function to supply missing values"
-msgstr "dict 的子類別，在值不存在 dict 當中時會呼叫一個產生函式。"
+msgstr "dict 的子類別，當值不存在 dict 中時會呼叫一個提供預設值的產生函式"
```
建議function的稱呼為「函數」或「函式」應統一

mattwang44 replied:
皆統一為「函式」

toto6038 commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:31
 msgid "wrapper around dictionary objects for easier dict subclassing"
-msgstr ""
+msgstr "dict 物件的包裝器 (wrapper)，簡化了 dict 的子類別化"
```
1. 微軟對wrapper的翻譯為「包裝函式」，個人無偏好何種稱呼，提供您參考
2. 「簡化了⋯⋯的子類別化」似乎有點拗口，也許可將「子類別化」改為「子類別過程」

mattwang44 replied:
1. 現有譯文 (tutorial/classes.po) 翻作 `包裝器`，我就先沿用了。未來若翻譯志工有一起制定出更完整的翻譯規則後，會再回頭檢視這些仍未有明確翻譯的內容！
2. updated

toto6038 commented on library/collections.po:
```diff
@@ -229,44 +270,61 @@ msgid ""
 #: ../../library/collections.rst:148
 msgid "This section shows various approaches to working with chained maps."
-msgstr ""
+msgstr "這一章節提供了多種操作 ChainMap 的案例。"
```
此章節提供了多種操作 ChainMap 的案例。

toto6038 commented on library/collections.po:
```diff
@@ -331,55 +395,62 @@ msgid ""
 #: ../../library/collections.rst:331
 msgid "This class method is not implemented for :class:`Counter` objects."
-msgstr ""
+msgstr "這個方法沒有在 :class:`Counter` 物件中被實作出來。"
```
此方法沒有（被）實作於 :class:`Counter` 物件中。

toto6038 commented on library/collections.po:
```diff
@@ -396,21 +471,29 @@ msgid ""
 #: ../../library/collections.rst:345
 msgid "Rich comparison operations were added."
-msgstr ""
+msgstr "增加了富比較運算。"
```
感覺硬著頭皮翻，好像也蠻奇怪。建議：豐富比較運算

mattwang44 replied:
我覺得我直接放原文好了，rich comparison 是 Python 專有名詞，找不除了 `富比較` 以外的譯文，但我根本沒聽過有人翻譯這個詞。

toto6038 commented on library/collections.po:
```diff
@@ -422,17 +505,22 @@ msgid ""
 "Unary addition and subtraction are shortcuts for adding an empty counter or "
 "subtracting from an empty counter."
-msgstr ""
+msgstr "加和減一元運算子分別是加上空的 Counter 和自空 Counter 減去的簡寫。"
```
加減法的一元運算子分別是加上空的 Counter、從空 Counter 減去的簡寫。

toto6038 commented on library/collections.po:
```diff
@@ -471,53 +567,69 @@ msgid ""
 "support addition, subtraction, and comparison."
 msgstr ""
+"Multiset 相關方法只為了處理正值而設計，其輸入允許是 0 或負值但只有正值會被輸"
+"出。型別上並無限制，但其值之型別必須支援加、減和比較操作。"
```
出。並無型別限制，但其值的型別須支援加、減及比較運算。

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 #: ../../library/collections.rst:489
 msgid "Create a shallow copy of the deque."
-msgstr ""
+msgstr "建立一個 deque 的淺複製。"
```
建立一個 deque 的淺複製（shallow copy）

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 "Extend the right side of the deque by appending elements from the iterable "
 "argument."
-msgstr ""
+msgstr "將 iterable 參數加入 deque 的右側。"
```
第 216 行處的 argument 譯作「引數」，是否有必要統一其稱呼？

mattwang44 replied:
此為翻譯錯誤，`argument` 應譯作 `引數`，而 `parameter` 才是 `參數`。

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 "Note, the series of left appends results in reversing the order of elements "
 "in the iterable argument."
 msgstr ""
+"將 iterable 參數加入 deque 的左側。要注意的是，加入後的元素順序和 iterable 參"
```
同第700行

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 "If the insertion would cause a bounded deque to grow beyond *maxlen*, an :"
 "exc:`IndexError` is raised."
 msgstr ""
+"如果這個插入動作會造成一個被限制長度的 deque 的長度超過 *maxlen* 的話，\\ :"
+"exc:`IndexError` 例外將被引發。"
```
如果此插入操作導致 deque 超過其長度上限 *maxlen*，會引發 exe:`IndexError`例外。

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 "Remove and return an element from the right side of the deque. If no "
 "elements are present, raises an :exc:`IndexError`."
 msgstr ""
+"移除 deque 的最右側元素並將其回傳，如果已經沒有任何元素則引發一個 :exc:"
+"`IndexError`\\ 。"
```
移除並回傳 deque 的最右側元素。若本來就沒有任何元素，則會引發 :exe:`IndexError`例外。

toto6038 commented on library/collections.po:
```diff
@@ -545,103 +663,121 @@ msgid ""
 "Remove and return an element from the left side of the deque. If no elements "
 "are present, raises an :exc:`IndexError`."
 msgstr ""
+"移除 deque 的最左側元素並將其回傳，如果已經沒有任何元素則引發一個 :exc:"
+"`IndexError`\\ 。"
```
同第737、738行

toto6038 commented on library/collections.po:
```diff
@@ -768,32 +948,39 @@ msgid ""
 #: ../../library/collections.rst:748
 msgid ":class:`defaultdict` objects support the following instance variable:"
-msgstr ""
+msgstr ":class:`defaultdict` 物件支援以下實例變量："
```
variable為變數

toto6038 commented on library/collections.po:
```diff
@@ -847,6 +1049,9 @@ msgid ""
 "are used, and they add the ability to access fields by name instead of "
 "position index."
 msgstr ""
+"Named tuple（具名元組）賦予 tuple 中各個位置意義，使程式碼更有可讀性與自我文"
+"件性。它們可以用於任何普通 tuple，添加透過名稱（而非位置索引）來存取欄位的能"
+"力。"
```
1. 「自我文件性」一詞不易理解，且將其省略不會影響句意。
2. 1053 行tuple後的「添加」，可改為「賦予其」

mattwang44 replied:
1. readability 和 self-documenting 意義不同，我不認為省略後會保持原意。這邊是參考台灣國家教育研究院辭書資訊網中最接近之譯文 `self-documenting code -> 自我文件碼`。
2. updated

josix replied:
我想也可以在自我文件性後加註（self-documenting）方便理解

toto6038 commented on library/collections.po:
```diff
@@ -966,88 +1202,112 @@ msgid ""
 "Return a new instance of the named tuple replacing specified fields with new "
 "values::"
 msgstr ""
+"回傳一個新的 named tuple 實例，並將指定欄位替換為新的值：\n"
```
本檔案前段已存在named tuple「具名元組」的稱呼

mattwang44 replied:
已於 `collections.rst:834` 附上譯文，後續皆放原文、不再翻譯（如同 ChainMap 之 `對映鏈結`、OrderedDict 之 `有序字典` 譯文都只會出現一次）。

toto6038 commented on library/collections.po:
```diff
@@ -966,88 +1202,112 @@ msgid ""
 "Docstrings can be customized by making direct assignments to the ``__doc__`` "
 "fields:"
-msgstr ""
+msgstr "透過直接賦值給 ``__doc__``，可以自定義說明文件字串："
```
建議customize翻譯為「自訂」、「客製化」

toto6038 commented on library/collections.po:
```diff
@@ -966,88 +1202,112 @@ msgid ""
 "tuples.  It also provides an elegant notation using the :keyword:`class` "
 "keyword::"
 msgstr ""
+"關於為 named tuple 新增型別提示的方法，請參閱 :class:`typing.NamedTuple`，它"
```
同第 1214 行

toto6038 commented on library/collections.po:
```diff
@@ -1081,33 +1348,41 @@ msgid ""
 msgid ""
 "The equality operation for :class:`OrderedDict` checks for matching order."
-msgstr ""
+msgstr ":class:`OrderedDict` 之相等性檢測會檢查順序是否相同。"
```
建議將 equality operation 譯作「相等性運算」

mattwang44 committed:
fix: update based on review comment

mattwang44 commented:
感謝 @toto6038！我已經根據 review comment 新增 commit 373ef73。

josix commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:25
 msgid "factory function for creating tuple subclasses with named fields"
-msgstr "用來建立一個欄位擁有名字的 tuple 子類別的函數"
+msgstr "用來建立具名欄位的 tuple 子類別的產生函式"
```
這邊會不會需要保留 factory function 的意思？”產生函式” 感覺可能沒有包含 design pattern 中 factory method 的意涵？

mattwang44 replied:
發現 glossary 裡面有翻譯成「工廠函式」，這邊就也改成一樣。

josix commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:31
 msgid "wrapper around dictionary objects for easier dict subclassing"
-msgstr ""
+msgstr "dict 物件的包裝器 (wrapper)，簡化了 dict 的子類別過程"
```
這邊 subclassing 可能可以翻成動作 “子類別化”
```suggestion
msgstr "dict 物件的包裝器 (wrapper)，簡化了 dict 的子類別化過程"
```

josix commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:32
 msgid "wrapper around list objects for easier list subclassing"
-msgstr ""
+msgstr "list 物件的包裝器，簡化了 list 的子類別過程"
```
Same as above.
```suggestion
msgstr "list 物件的包裝器，簡化了 list 的子類別化過程"
```

josix commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:33
 msgid "wrapper around string objects for easier string subclassing"
-msgstr ""
+msgstr "字串物件的包裝器，簡化了字串的子類別過程"
```
Same as above.
```suggestion
msgstr "字串物件的包裝器，簡化了字串的子類別化過程"
```

josix commented on library/collections.po:
```diff
@@ -132,33 +142,45 @@ msgid ""
 "mapping."
 msgstr ""
+"一個 :class:`ChainMap` 將多個 dict 或其他映射組合在一起，建立一個獨立、可更新"
+"的視野。如果沒有指定 *maps*，預設會提供一個空字典讓每個新鏈結都至少有一個映"
```
View 會不會翻成視圖會比較好？看起來和 MVC 的 view 雷同負責資料顯示的職責

mattwang44 replied:
modified

josix commented on library/collections.po:
```diff
@@ -132,33 +142,45 @@ msgid ""
 "if one of the underlying mappings gets updated, those changes will be "
 "reflected in :class:`ChainMap`."
 msgstr ""
+":class:`ChainMap` 透過指標將底層映射合併，所以當一個底層映射被更新，這些改變"
```
我想 reference 應該要翻成參照會比較好一些，python 中應該沒有 potienr？
```suggestion
":class:`ChainMap` 透過參照將底層映射合併，所以當一個底層映射被更新，這些改變"
```

josix reviewed (changes requested):
抱歉我晚 review 了，先 review 了 ChainMap 的部分，再麻煩 @mattwang44 檢查，我後續會再補上其他部分，感謝！

mattwang44 committed:
Apply suggestions from code review

Co-authored-by: Josix <wilson8507@gmail.com>

mattwang44 committed:
fix(library/collections): update translation of the term `mapping`

Unified the translation of the term `mapping` from `映射` to `對映`, as it has been translated in glossary.

josix commented on library/collections.po:
```diff
@@ -331,55 +395,62 @@ msgid ""
 "encountered in the left operand and then by the order encountered in the "
 "right operand."
 msgstr ""
+"作為 :class:`dict` 的子類別，\\ :class:`Counter` 繼承了記憶插入位置的功能。"
```
```suggestion
"作為 :class:`dict` 的子類別，\\ :class:`Counter` 繼承了記憶插入順序的功能。"
```

mattwang44 replied:
updated

josix commented on library/collections.po:
```diff
@@ -331,55 +395,62 @@ msgid ""
 msgstr ""
-"回傳一個每個元素都重複出現計算次數的 iterator 物件，其中元素的排列順序會是隨"
-"機的。如果元素的出現次數小於 1 ，:meth:`elements` 方法會忽略這些元素。"
+"回傳每個元素都重複出現計算次數的 iterator 物件，其中元素的回傳順序是依照各元"
```
iterator 不確定有沒有已經翻譯成 "迭代器"？可能要統一一下

toto6038 replied:
依[國教院](https://terms.naer.edu.tw/detail/17008148/)資料，iterator 中文稱呼為「疊代器」，「『迭』代器」為誤用。
在黃敬群教授前幾日的[講座影片](https://youtu.be/pTcRq__iKzI?t=8764)中，聊天室亦有稍微討論，連結已標註時間碼。

mattwang44 replied:
已補上原文

josix replied:
感謝 @toto6038 補充!

josix reviewed (changes requested):
Finish reviewing the `Counter` part.

mattwang44 committed:
fix: correct `迭代` as `疊代`

toto6038 commented on library/collections.po:
```diff
@@ -34,83 +37,85 @@ msgid ""
 #: ../../library/collections.rst:27
 msgid "dict-like class for creating a single view of multiple mappings"
-msgstr "一個像是 dict 的類別，用來為多個 mapping 建立單一的 view 。"
+msgstr "一個類似 dict 的類別，用來為多個對映 (mapping) 建立單一的視圖 (view)"
```
view 是否應譯作「視野」較佳？
參考資料：[國家教育研究院](http://terms.naer.edu.tw/detail/1288562/)

mattwang44 replied:
我原先的確是查國教院所以譯作「視野」，但 @josix 之前的 [review comment](https://github.com/python/python-docs-zh-tw/pull/141#discussion_r777358808) 提到這邊比較像是指 MVC 中 view 的概念，此義普遍會被翻作「視圖」（例如[維基百科](https://zh.wikipedia.org/wiki/MVC)，但還找不到比較有公信力的機構為這個詞作翻譯），而國教院對該詞的解釋（`在關聯式資料庫中描述使用者之子綱目的用語。`）感覺並不相符，所以才改成「視圖」。

mattwang44 replied:
也許以下兩處的「檢視」也應該改成「視圖」？

https://github.com/python/python-docs-zh-tw/blob/2261735da57efc6c575995cbc952bb42d975bb22/library/collections.po#L1433-L1439
https://github.com/python/python-docs-zh-tw/blob/2261735da57efc6c575995cbc952bb42d975bb22/library/stdtypes.po#L4834-L4836

josix replied:
嗯嗯對，我覺得可以改成"視圖"

mattwang44 replied:
已更正囉～

toto6038 commented on library/collections.po:
```diff
@@ -132,33 +142,45 @@ msgid ""
 "In contrast, writes, updates, and deletions only operate on the first "
 "mapping."
 msgstr ""
+"搜索 (lookup) 陸續查詢底層對映，直到鍵被找到，然而讀取、更新和刪除就只會對第"
```
lookup 可譯為「檢索」

mattwang44 replied:
modified

toto6038 commented on library/collections.po:
```diff
@@ -526,6 +638,9 @@ msgid ""
 "approximately the same O(1) performance in either direction."
 msgstr ""
+"Deque（發音為 \"deck\"，為 \"double-ended queue\" 的簡稱）為 stack 和 queue "
+"的推廣。deque 支援執行緒安全 (thread-safe)，且能夠高記憶效率地在頭和尾加入和"
```
「且能夠高記憶效率地在頭和尾加入...」->「且能夠節省記憶體地在頭尾加入...」

josix commented on library/collections.po:
```diff
@@ -526,6 +638,9 @@ msgid ""
 "approximately the same O(1) performance in either direction."
 msgstr ""
+"Deque（發音為 \"deck\"，為 \"double-ended queue\" 的簡稱）為 stack 和 queue "
+"的推廣。deque 支援執行緒安全 (thread-safe)，且能夠高記憶效率地在頭和尾加入和"
```
```suggestion
"的一般化。deque 支援執行緒安全 (thread-safe)，且能夠有效率地節省記憶體在頭和尾加入和"
```

mattwang44 replied:
modified

josix commented on library/collections.po:
```diff
@@ -651,36 +787,49 @@ msgid ""
 "to access the first element.  Indexed access is O(1) at both ends but slows "
 "to O(n) in the middle.  For fast random access, use lists instead."
 msgstr ""
+"除了以上使用方式，deque 亦支援了疊代、pickle（封存）、``len(d)``、"
```
pickle 這個命名源自 Modula-3 這個語言取自醃製保留風味的意思，我想 pickle 可能可以不用翻譯。[reference here](https://stackoverflow.com/questions/27324986/pickles-why-are-they-called-that)

mattwang44 replied:
已更正，也在 README 裡面加註 pickle 可以保留原文就好

josix reviewed (changes requested):
Finish reviewing `deque` part.

mattwang44 committed:
Apply suggestions from code review

Co-authored-by: Josix <wilson8507@gmail.com>
Co-authored-by: toto6038

mattwang44 committed:
doc: add `pickle` to terms that doesn't have to be translated

josix commented on library/collections.po:
```diff
@@ -728,38 +895,50 @@ msgid ""
 "to provide a default value for the given *key*, this value is inserted in "
 "the dictionary for the *key*, and returned."
 msgstr ""
+"如果 :attr:`default_factory` 不為 ``None``，它會被呼叫（不帶引數）來為給定的 "
```
```suggestion
"如果 :attr:`default_factory` 不為 ``None``，它會不帶引數地被呼叫來為給定的 "
```

josix commented on library/collections.po:
```diff
@@ -806,13 +992,21 @@ msgid ""
+"當每個鍵第一次被存取時，它還沒有存在於對映中，所以會自動呼叫 :attr:"
+"`~defaultdict.default_factory` 方法來回傳一個空的 :class:`list` 以建立一個條"
+"目，\\ :meth:`list.append` 操作後續會再新增值到這個新的列表裡。當再次存取該鍵"
+"時，就如普通字典般操作（回傳該鍵所指到的 list），\\ :meth:`list.append` 也會"
```
nit. 可能改成對應會符合 mapping 意涵
```suggestion
"時，就如普通字典般操作（回傳該鍵所對應到的 list），\\ :meth:`list.append` 也會"
```

josix reviewed (changes requested):
Finish reviewing defaultdict

mattwang44 committed:
Apply suggestions from code review

Co-authored-by: Josix <wilson8507@gmail.com>

mattwang44 committed:
fix: correct some mistranslated strings

josix commented on library/collections.po:
```diff
@@ -847,6 +1048,9 @@ msgid ""
 "are used, and they add the ability to access fields by name instead of "
 "position index."
 msgstr ""
+"Named tuple（具名元組）賦予 tuple 中各個位置意義，使程式碼更有可讀性與自我文"
```
想確認下 tuple 目前有統一翻譯成元組嗎？還是保留 tuple 原文即可，假如有統一的話記得再更新 glossary，感謝。

mattwang44 replied:
readme 有提到建議保留原文，所以在首次出現的地方（intro 部份）有附上譯文，其他地方就直接放原文

josix replied:
瞭解，感謝說明🙏

josix commented on library/collections.po:
```diff
@@ -847,6 +1048,9 @@ msgid ""
 "position index."
 msgstr ""
+"Named tuple（具名元組）賦予 tuple 中各個位置意義，使程式碼更有可讀性與自我文"
+"件性。它們可以用於任何普通 tuple，賦予其透過名稱（而非位置索引）來存取欄位的"
```
```suggestion
"件性。它們可以用於任何普通 tuple 可使用的場合，賦予其透過名稱（而非位置索引）來存取欄位的"
```

josix commented on library/collections.po:
```diff
@@ -856,13 +1060,20 @@ msgid ""
 msgstr ""
+"回傳一個名為 *typename* 的新 tuple 子類別。這個新的子類別被用於建立類似 "
+"tuple 的物件，可以透過屬性名稱來存取欄位，它同時也是可索引 (indexable) 和可疊"
+"代的 (iterable)。子類別實例同樣有文件字串（有類別名 typename 和欄位名 "
```
```suggestion
"代的 (iterable)。子類別實例同樣有文件字串（docstring）（有類別名 typename 和欄位名 "
```

josix commented on library/collections.po:
```diff
@@ -889,75 +1107,90 @@ msgid ""
 msgid ""
 "Class method that makes a new instance from an existing sequence or iterable."
-msgstr ""
+msgstr "從已存在的序列或可疊代物件建立一個新實例。"
```
```suggestion
msgstr "從已存在的序列或可疊代物件建立一個新實例的類別方法。"
```

josix reviewed (changes requested):
Finish reviewing `namedtuple`

mattwang44 committed:
Apply suggestions from code review

Co-authored-by: Josix <wilson8507@gmail.com>

josix commented on library/collections.po:
```diff
@@ -1081,33 +1345,41 @@ msgid ""
 "The :meth:`popitem` method of :class:`OrderedDict` has a different "
 "signature.  It accepts an optional argument to specify which item is popped."
 msgstr ""
+":class:`OrderedDict` 類別的 :meth:`popitem` 方法有不同的 signature，它接受傳"
```
不確定signature是不是需要翻，[wiki 是翻成函數簽名](https://zh.wikipedia.org/wiki/%E5%87%BD%E6%95%B0%E5%8E%9F%E5%9E%8B)

mattwang44 replied:
modified
有填 glossary 擴充表單，之後可以來討論一下這個翻譯

josix commented on library/collections.po:
```diff
@@ -1116,19 +1388,30 @@ msgid ""
 "out)` order if *last* is true or :abbr:`FIFO (first-in, first-out)` order if "
 "false."
 msgstr ""
+"Ordered dictionary 的 :meth:`popitem` 方法移除並回傳一個 (key, value) 鍵值"
```
```suggestion
"Ordered dictionary 的 :meth:`popitem` 方法移除並回傳一個鍵值（key, value）"
```

josix commented on library/collections.po:
```diff
@@ -1181,6 +1478,9 @@ msgid ""
 "subclass directly from :class:`dict`; however, this class can be easier to "
 "work with because the underlying dictionary is accessible as an attribute."
 msgstr ""
+":class:`UserDict` 類別 dict 物件的包裝器。因為已經可以直接自 :class:`dict` 建"
```
```suggestion
":class:`UserDict` 類別是作為 dict 物件的包裝器。因為已經可以直接自 :class:`dict` 建"
```

josix commented on library/collections.po:
```diff
@@ -1190,21 +1490,25 @@ msgid ""
 msgstr ""
+"模擬字典的類別。實例的內容被存於一個字典，可透過 :class:`UserDict` 的 :attr:"
+"`data` 屬性來做存取。如果有提供 *initialdata*\\ ，\\ :attr:`data` 屬性會被初"
+"始化為其值；要注意指到 *initialdata* 的指標不會被保留，使其可被用於其他目的。"
```
```suggestion
"始化為其值；要注意指到 *initialdata* 的參照不會被保留，使其可被用於其他目的。"
```

josix commented on library/collections.po:
```diff
@@ -1259,10 +1578,12 @@ msgid ""
 "consult the sources for information about the methods which need to be "
 "provided in that case."
 msgstr ""
+"如果希望一個自此獲得的子類別能夠別按照上述條件，那所有該類別支援的特殊方法則"
```
```suggestion
"如果希望一個自此獲得的子類別不遵從上述要求，那所有該類別支援的特殊方法則"
```

josix commented on library/collections.po:
```diff
@@ -1271,6 +1592,9 @@ msgid ""
 "directly from :class:`str`; however, this class can be easier to work with "
 "because the underlying string is accessible as an attribute."
 msgstr ""
+":class:`UserString` 類別是字串物件的包裝器 因為已經可以從 :class:`str` 直接建"
```
```suggestion
":class:`UserString` 類別是字串物件的包裝器，因為已經可以從 :class:`str` 直接建"
```

josix reviewed (changes requested):
Finish reviewing rest parts.

mattwang44 committed:
Apply suggestions from code review

Co-authored-by: Josix <wilson8507@gmail.com>

mattwang44 committed:
chore: sync with cpython and fix fuzzy entry

josix reviewed (approved):
Let's merge it! Thanks @mattwang44 and @toto6038 for working on this.
